plugins/maciVoting/services/coordinator/utils/auth.ts

```diff
@@ -0,0 +1,52 @@
+import fs from "fs";
+import { type KeyLike, publicEncrypt } from "crypto";
+import { type Signer, getBytes } from "ethers";
+import { hashMessage } from "viem";
+import { ErrorCodes } from "./errors";
+
+/**
+ * @noitce encrypt copied from https://github.com/privacy-scaling-explorations/maci/blob/dev/apps/coordinator/ts/crypto/crypto.service.ts
+ * Converted from class method to function
+ *
+ * Encrypt plaintext with public key
+ *
+ * @param publicKey - public key
+ * @param value - plaintext
+ * @returns ciphertext
+ */
+const encrypt = (publicKey: KeyLike, value: string): string => {
+  try {
+    const encrypted = publicEncrypt(publicKey, Buffer.from(value));
+
+    return encrypted.toString("base64");
+  } catch (error) {
+    throw new Error(`Error code: ${ErrorCodes.ENCRYPTION.toString()}, Error: ${error}`);
+  }
+};
+
+/**
+ * @noitce encryptWithCoordinatorRSAPublicKey copied from https://github.com/privacy-scaling-explorations/maci/blob/dev/apps/coordinator/tests/utils.ts
+ * Uses encrypt above function rather than calling CryptoService class
+ *
+ * Encrypt a message using the coordinator's public key
+ * @param message to encrypt
+ * @returns encrypted message (ciphertext)
+ */
+export const encryptWithCoordinatorRSAPublicKey = async (message: string): Promise<string> => {
```

Comment:
This function is not required. The publicKey should be fetch from the coordinator service using the endpoint `/proof/publicKey` and then we should send that publicKey to `encryptWithCoordinatorRSA` (the function above)

Reply:
Removed